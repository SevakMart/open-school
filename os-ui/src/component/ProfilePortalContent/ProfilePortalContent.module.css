.mainContent {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: none;
  border: solid;
  cursor: pointer;
}
.mainContent .iconContainer, .mainContent .titleContent {
  margin: 0 24px 0;
}/*# sourceMappingURL=ProfilePortalContent.module.css.map */