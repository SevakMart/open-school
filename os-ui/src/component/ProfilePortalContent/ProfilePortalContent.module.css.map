{"version":3,"sources":["ProfilePortalContent.module.scss","../../styles/Mixins/FlexMixins.scss","ProfilePortalContent.module.css"],"names":[],"mappings":"AAEA;ECDE,aAAA;EACA,mBDCqB;ECArB,mBAAA;EACA,qBDDgC;EAC9B,aAAA;EACA,eAAA;AEEJ;AFDI;EACI,gBAAA;AEGR","file":"ProfilePortalContent.module.css"}