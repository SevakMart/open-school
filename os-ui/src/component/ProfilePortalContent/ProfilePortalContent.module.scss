@import "../../styles/Mixins/FlexMixins.scss";

.mainContent {
    @include flexStyle(row,center,none);
    border:solid;
    cursor:pointer;
    .iconContainer, .titleContent{
        margin:0 24px 0;
    }
}