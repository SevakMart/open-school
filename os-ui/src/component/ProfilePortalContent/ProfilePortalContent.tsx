import { useNavigate } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { useContext } from 'react';
import { removeUserInfoFromLocalStorage } from '../../redux/Slices/UserInfoSlice';
import styles from './ProfilePortalContent.module.scss';
import { signInContext } from '../../contexts/Contexts';

const ProfilePortalContent = ({ icon, children, isSignOut }:any) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const { setSignIn } = useContext(signInContext);
  const { mainContent, iconContainer, titleContent } = styles;

  const handleClick = () => {
    if (isSignOut) {
      dispatch(removeUserInfoFromLocalStorage());
      navigate('/');
      setSignIn(false);
    }
  };

  return (
    <div className={mainContent} onClick={handleClick}>
      <div className={iconContainer}>
        {icon}
      </div>
      <p className={titleContent}>{children}</p>
    </div>
  );
};
export default ProfilePortalContent;
