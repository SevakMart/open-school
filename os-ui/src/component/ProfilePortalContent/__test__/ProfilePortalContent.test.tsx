import { render } from '@testing-library/react';
import ProfilePortalContent from '../ProfilePortalContent';

describe('Create unit tests for ProfilePortalContent component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <ProfilePortalContent
        isSignOut={false}
        children={<h2>Hello world</h2>}
        icon={<p>{'>'}</p>}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
