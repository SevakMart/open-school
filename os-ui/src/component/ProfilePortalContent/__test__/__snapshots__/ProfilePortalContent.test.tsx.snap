// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create unit tests for ProfilePortalContent component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContent"
  >
    <div
      class="iconContainer"
    >
      <p>
        &gt;
      </p>
    </div>
    <p
      class="titleContent"
    />
    <h2>
      Hello world
    </h2>
    <p />
  </div>
</DocumentFragment>
`;
